File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Effect/Combat/EntangleEffect.ts
import { FairyGUI } from "csharp";
import { View_EntangleEffect } from "../../Gen/View/ModTheSpire_Effect";
import { AbstractEffect } from "../AbstractEffect";

export class EntangleEffect extends AbstractEffect{
    private Loader: FairyGUI.GComponent;
    public constructor(private _X:number,private _Y:number,private _TargetX:number,private _TargetY:number) {
        super();
        this.Loader = <FairyGUI.GComponent>View_EntangleEffect.CreateInstance();
        this.Loader.SetPosition(_X,_Y,0);
        let transition = this.Loader.GetTransition("t0");
        transition.SetValue("XY",_TargetX,_TargetY);
        transition.Play();
    }
    
    Update() {
        super.Update();
        if(this.Duration >= 1){
            this.IsDone = true;
            this.Loader.Dispose();
        }
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/EntangleEffect.cs
using FairyGUI;
using ModTheSpire.ModTheSpire_Effect;

namespace ModTheSpire.Effect.Combat
{
    public class EntangleEffect : AbstractEffect
    {
        private GComponent mLoader;
        public EntangleEffect(float x, float y, float targetX, float targetY)
        {
            mLoader = View_EntangleEffect.CreateInstance();
            mLoader.SetPosition(x,y,0);
            var transition = mLoader.GetTransition("t0");
            transition.SetValue("XY", targetX, targetX);
            transition.Play();
        }

        public override void Update()
        {
            base.Update();
            if(Duration >= 1)
                Finish();
        }

        public override void Dispose()
        {
            base.Dispose();
            mLoader.Dispose();
        }
    }
}